tests/functional/views/test_legacy_publishing.py

```diff
@@ -1,366 +1,432 @@
+import unittest
+from pyramid import testing
 from lxml import etree
 
 from litezip.main import COLLECTION_NSMAP
 
+class TestBypassingAuth():
+    def setup(self):
+        request = testing.DummyRequest()
+        request.context = testing.DummyResource()
+        # import pdb; pdb.set_trace()
+        self.config = testing.setUp(request=request)
+        self.config.testing_securitypolicy(userid='userid', permissive=True)
```

Comment:
Probably best to either use the existing `user1` as the tests publish as. Ideally the publisher field will now be dropped since the user id is available. There should be a todo/fixme to this effect somewhere in the view.

Reply:
Now that I think about it, this is likely the wrong approach to this... The app is already configured as part of the `webapp` fixture. So the configuring you are doing here is actually creating a secondary app (`Configurator` object, registry, etc). The pyramid testing stuff is more unit-testing oriented rather than functional tests. Sorry I didn't mention that earlier, I wasn't fully cognizant when we discussed this last.

Take a look at http://webtest.pythonpaste.org/en/latest/testapp.html#modifying-the-environment-simulating-authentication for how to do this with WebTest, which is what the `webapp` fixture uses.

Reply:
Yup, that's exactly how Bryan fixed it.